archive/imo/imo2014_q1.lean

```diff
@@ -0,0 +1,183 @@
+/-
+Copyright (c) 2021 Adrián Doña Mateo. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Adrián Doña Mateo
+-/
+
+import algebra.big_operators
+import data.pnat.basic
+
+/-!
+# IMO 2014 Q1
+
+Let a₀ < a₁ < a₂ < ⋯ be an infinite sequence of positive integers.
+Prove that there exists a unique n ≥ 1 such that
+
+  aₙ < (a₀ + a₁ + ⋯ + aₙ) / n < aₙ₊₁.
+
+This solutions is a translation of the official solution, which may be found as the
+solution to problem A1 [here](https://www.imo-official.org/problems/IMO2014SL.pdf).
+-/
+
+open_locale big_operators
+
+noncomputable theory
+
+/- The integer sequence. -/
+variable (a : ℕ → ℤ)
+
+/- We define an auxiliary sequence `d a : ℕ+ → ℤ` by
```

Comment:
If you start your comment with `/--` it becomes a docstring, which shows up on mouse-over, whenever some places the cursor over `d` anywhere below (in VScode).